File: MS.NET/Fundamentals/Generics/Basics/genclasstest.cs
using System;

public class SimpleStack<V>
{
	class Node
	{
		internal V value;
		internal Node below;

		internal Node(V v, Node n)
		{
			value = v;
			below = n;
		}
	}

	private Node top;

	public void Push(V element)
	{
		top = new Node(element, top);
	}

	public V Pop()
	{
		V element = top.value;
		top = top.below;
	
		return element;
	}

	public bool Empty()
	{
		return top == null;
	}
}

static class Program
{
	public static void Main()
	{
		SimpleStack<string> a = new SimpleStack<string>();
		a.Push("monday");
		a.Push("tuesday");
		a.Push("wednesday");
		a.Push("thursday");
		a.Push("friday");
		//a.Push(12.3);

		SimpleStack<string> b = new SimpleStack<string>();
		b.Push("June");
		b.Push("May");
		b.Push("April");
		b.Push("March");

		var c = new SimpleStack<Interval>();
		c.Push(new Interval(4, 31));
		c.Push(new Interval(7, 42));
		c.Push(new Interval(6, 23));
		c.Push(new Interval(3, 54));
		c.Push(new Interval(5, 15));

		var d = new SimpleStack<object>();
		d.Push("saturday");
		d.Push(new Interval(2, 30));
		d.Push(12.3);

		while(!a.Empty())
			Console.WriteLine(a.Pop());
		Console.WriteLine("----------------------");

		while(!b.Empty())
			Console.WriteLine(b.Pop());
		Console.WriteLine("----------------------");

		while(!c.Empty())
			Console.WriteLine(c.Pop());
		Console.WriteLine("----------------------");

		while(!d.Empty())
			Console.WriteLine(d.Pop());
		
	}
}

File: Java/Fundamentals/Generic/Basics/GenClassTest.java
class GenClassTest{
	
	static class SimpleStack<V>{
		
		private Node top;

		class Node{
			V value;
			Node below;

			Node(V entry){
				value = entry;
				below = top;
			}
		}

		public void push(V item){
			top = new Node(item);
		}

		public V pop(){
			V item = top.value;
			top = top.below;
			return item;
		}

		public boolean empty(){
			return top == null;
		}
	}

	public static void main(String[] args){
		SimpleStack<String> a = new SimpleStack<String>();
		a.push("monday");
		a.push("tuesday");
		a.push("wednesday");
		a.push("thursday");
		a.push("friday");
		//a.push(4.56);
		SimpleStack<String> b = new SimpleStack<>();
		b.push("june");
		b.push("may");
		b.push("april");
		b.push("march");
		SimpleStack<Interval> c = new SimpleStack<>();
		c.push(new Interval(5, 31));
		c.push(new Interval(3, 52));
		c.push(new Interval(7, 43));
		c.push(new Interval(6, 14));
		c.push(new Interval(4, 25));
		SimpleStack<Object> d = new SimpleStack<>();
		d.push(new Interval(2, 30));
		d.push("sunday");
		d.push(4.56);
		while(!a.empty())
			System.out.println(a.pop());
		System.out.println("----------------------");
		while(!b.empty())
			System.out.println(b.pop());
		System.out.println("----------------------");
		while(!c.empty())
			System.out.println(c.pop());
		System.out.println("----------------------");
		while(!d.empty())
			System.out.println(d.pop());
	}
}





